enhancements/console/custom-logos.md

```diff
@@ -0,0 +1,147 @@
+
+### User Stories
+
+* As an OpenShift web console user, I want to be able to add custom logos for light and dark theme modes in the masthead and favicon.
+
+### Goals
+
+This feature should allow users to add custom logos for the masthead and favicon that are compatible with both light and dark theme modes in the OpenShift web console.
+
+### Non-Goals
+
+
+## Proposal
+
+### Workflow Description
+
+├── spec
+│   ├── customization
+│       ├── customLogos
+│           ├── type
+│           ├── themes
+│               ├── type
+│               ├── file
+│                   ├── key
+│                   ├── name
+└── ...
+
+### Risks and Mitigations
+
+1. Users could set both the `CustomLogoFile` and `CustomLogos` APIs. The `CustomLogos` API configuration will take precedence over the old `CustomLogoFile` field.
+
+2. If only the dark or light theme is specified, the default OCP logo will be used for the unspecified theme.
+
+3. Handling different custom logos for light and dark themes may increase the complexity of the backend and UI logic. Tests are added to validate the new  logic and a fallback is implemented to default to the OCP logo if any theme is missing.
+
+4. Users might experience confusion with the introduction of new logo configuration options, especially if they are familiar with the old method, which may soon be deprecated. Provide a comprehensive documentation and tools that guide users through the transition. Include clear instructions about the changes and their benefits.
+
+
+### Drawbacks
+
+N/A
+
+## API Design Details
+
+The configuration for custom logos will include support for both masthead and favicon types, with separate files for light and dark themes:
+
+```yaml
+customLogos:
+  - type: Masthead
+    themes:
+      - type: Light
+        file:
+          key: logo-light.svg
+          name: masthead-logo-light
+      - type: Dark
+        file:
+          key: logo-dark.svg
+          name: masthead-logo-dark
+  - type: Favicon
+    themes:
+      - type: Light
+        file:
+          key: favicon-light.ico
+          name: favicon-logo-light
+      - type: Dark
+        file:
+          key: favicon-dark.ico
+          name: favicon-logo-dark
+```
+
+### Attributes Description
+
+#### customLogos
+- `type`: Enumerate which specifies the type of custom logo. Available custom logo types are `Masthead` and `Favicon`.
+- `themes`: A list of themes for which the custom logo is defined.
+
+#### themes
+- `type`: Enumerate which specifies the type of theme. Available theme types are `Light`, `Dark` and `Default`.
+- `file`: Contains the file details for the custom logo.
+
+#### file
+- `key`: The key or path to the custom logo file.
+- `name`: The name of the ConfigMap containing the custom logo file.
+
+### Test Plan
```

Comment:
Lets list tests which we will deliver:
- unit tests in the API repo
- unit and e2e tests in the console-operator repo
- e2e tests in the console repo